src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrState = [];
+  let newState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
+      newState = {};
+      arrState.push({ ...newState });
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => delete newState[key]);
+      arrState.push({ ...newState });
```

Comment:
[CODE STYLE] - As per the previous comment, avoid repeating the push operation inside each condition. Move this operation outside and after the switch statement.